sway/config/output.c

```diff
@@ -142,96 +207,45 @@ static void merge_output_config(struct output_config *dst, struct output_config
 	}
 }
 
-static void merge_wildcard_on_all(struct output_config *wildcard) {
-	for (int i = 0; i < config->output_configs->length; i++) {
-		struct output_config *oc = config->output_configs->items[i];
-		if (strcmp(wildcard->name, oc->name) != 0) {
-			sway_log(SWAY_DEBUG, "Merging output * config on %s", oc->name);
-			merge_output_config(oc, wildcard);
-		}
-	}
-}
-
-static void merge_id_on_name(struct output_config *oc) {
-	struct sway_output *output = all_output_by_name_or_id(oc->name);
-	if (output == NULL) {
-		return;
-	}
-
-	const char *name = output->wlr_output->name;
+void store_output_config(struct output_config *oc) {
 	char id[128];
-	output_get_identifier(id, sizeof(id), output);
-
-	char *id_on_name = format_str("%s on %s", id, name);
-	if (!id_on_name) {
-		return;
+	bool merged = false;
+	bool wildcard = strcmp(oc->name, "*") == 0;
+	struct sway_output *output = wildcard ? NULL : output_by_name_or_id(oc->name);
+	if (!output && !wildcard) {
+		// There is no config by this name, just add it in
+		goto done;
 	}
```

Comment:
Hm, why do we need to grab the sway_output here? Won't the loop below do the right thing regardless?

Reply:
We need to know whether the output config matches an output's identifier or name, which requires us to look up the output. This info is needed to know which configs to supersede, with less specific configs wiping out settings in more specific configs in order for them to take effect.

I see that I had lost the call to `output_get_identifier` in a refactor, added back in.